image_metadata_script.py

```diff
@@ -0,0 +1,53 @@
+import os, time
+import filetype
+import csv
+import sys
+
+
+def image_metadata(path):
+
+    minio_link = 'None'
+    dataset = 'None'
+    is_processed = 0
+
+    try:
+        output_file = open('file_name.csv', 'w')
+
+        fields = ['IV_Name', 'Camera_ID', 'IV_date', 'IV_time', 'File_type', 'File_size', 'Minio_link', 'Dataset',
+                  'is_processed']
+        writer = csv.writer(output_file)
+        writer.writerow(fields)
+
+        try:
+            dirs = os.listdir(path)
+        except Exception as e:
+            print("Path is not a valid directory")
+            sys.exit(1)
+
+        for subdir in dirs:
+            path2 = os.path.join(path, subdir)
+            if os.path.isfile(path2):
+                continue
+            for f in os.listdir(path2):
+                curr_path = os.path.join(path2, f)
+                if os.path.isdir(curr_path):
+                    continue
+                st = os.stat(curr_path)
+                file_size = st.st_size
+                file_date = time.strftime('%Y.%m.%d', time.localtime(os.path.getmtime(curr_path)))
+                file_time = time.strftime('%H:%M:%S', time.localtime(os.path.getmtime(curr_path)))
+
+                ftype = filetype.guess(curr_path)
+                if ftype:
+                    file_type = ftype.extension
+
+                writer.writerow([f, subdir, file_date, file_time, file_type, file_size, minio_link, dataset,
```

Comment:
is f and subdir the correct fields to fill in? I thought they should be IV_name, which is curr_path, and cameraID

Reply:
f is the name of the image file which is a concatenation of the Camera ID, date, time, size and type. An example of file name is 5b17e02f811d4a000470c92d_2019-03-31_22-51-20-061917.png. This is the name of the images generated inside the 'results' folder which was created when I first called the ImageArchiver to get images from the database. Because this is how images were named inside the subdirectories inside the 'results' folder, I thought it is working fine. The curr_path displays the path to that image file. 
As for camera ID, I used subdir because each subdirectory inside the results folder is named after the Camera ID. 
To make it more clear, the output of the csv file is something like this:

![image](https://user-images.githubusercontent.com/32406375/57185510-c293dd00-6e9a-11e9-8a2b-5747b69bebb2.png)